### ISSUE TITLE
Unexpected behavior with `deepcopy` on `MutableList`

### ISSUE THREAD

**Issue by petriggles** [sqlalchemy/sqlalchemy#4603]
When performing a `deepcopy` on some serialized data, I found that anything that's of the type `MutableList` gets its elements duplicated. I'm using `1.3.2`

```python
>>> foo = MutableList(['bar'])
>>> print(foo)
['bar']
>>> print(deepcopy(foo))
['bar', 'bar']
```

**Comment by zzzeek** [comment]
hi -

can you not use deepcopy() for now?  I think deepcopy is kind of crazy to use ever.

**Comment by zzzeek** [comment]
First copy() calls our `__setstate__`, where we as instructed rebuild the list:

```
 /home/classic/.venv3/lib/python3.7/copy.py(282)_reconstruct()
-> y.__setstate__(state)
> /home/classic/dev/sqlalchemy/lib/sqlalchemy/ext/mutable.py(780)__setstate__()
-> self[:] = state

```

then it calls append() again?   how is that supposed to work

```
> /home/classic/.venv3/lib/python3.7/copy.py(300)_reconstruct()
-> for item in listiter:
(Pdb) next
> /home/classic/.venv3/lib/python3.7/copy.py(301)_reconstruct()
-> y.append(item)
(Pdb) !y
[1, 2]
(Pdb) !item
1
(Pdb) 

```

**Comment by sqla-tester** [comment]
**Mike Bayer** has proposed a fix for this issue in the **master** branch:

**Add __reduce_ex__ to MutableList; add compat for older pickles** https://gerrit.sqlalchemy.org/1186

**Comment by petriggles** [comment]
>can you not use deepcopy() for now? I think deepcopy is kind of crazy to use ever.

deepcopy is a great fit for my use case - I was on `1.1.18` and ran into this issue while looking to upgrade, so I can stick on that version for now

**Comment by zzzeek** [comment]
the fix will be out in 1.3.3.    i had to change the pickle format of MutableList but fortunately the old formats can still unpickle.
